Yosemite/Yosemite/Tools/POS/POSOrderService.swift

```diff
@@ -66,73 +60,17 @@ public final class POSOrderService: POSOrderServiceProtocol {
     }
 }
 
-private struct POSOrderSyncProductType: OrderSyncProductTypeProtocol, Hashable {
-    let productID: Int64
-    let price: String
-    // Not used in POS but have to be included for the app usage.
-    let productType: ProductType
-    let bundledItems: [ProductBundleItem]
-
-    init(productID: Int64, price: String, productType: ProductType, bundledItems: [ProductBundleItem] = []) {
-        self.productID = productID
-        self.price = price
-        self.productType = productType
-        self.bundledItems = bundledItems
-    }
-}
-
-private extension POSOrderService {
-    func updateOrder(_ order: Order, cart: POSCart) -> Order {
-        // Removes all existing items by setting quantity to 0.
-        let itemsToRemove = order.items.compactMap {
-            Self.removalProductInput(item: $0)
-        }
-
-        // Adds items from the latest cart grouping by item.
-        let itemsToAdd = cart.items.createGroupedOrderSyncProductInputs().values
-        let itemsToSync = itemsToRemove + itemsToAdd
-
-        var order = ProductInputTransformer.updateMultipleItems(with: itemsToSync, on: order, shouldUpdateOrDeleteZeroQuantities: .update)
-        order = updateCoupons(cart.coupons, on: order)
-
-        return order
+private extension Order {
+    func addItems(_ cartItems: [POSCartItem]) -> Order {
+        let itemsToAdd = Array(cartItems.createGroupedOrderSyncProductInputs().values)
```

Comment:
Do we need to call `sanitizingLocalItems` on these, or is that only relevant for updates?

Reply:
Ah, glad I spotted it then! I wasn't sure because it could have been something only required on update... though, our products are all added as an update, aren't they? Rather than in the initial POST request?

Reply:
> though, our products are all added as an update, aren't they? Rather than in the initial POST request?

No. We add them as initial request when creating an order. `createGroupedOrderSyncProductInputs` turns products into order items. `sanitize` makes sure totals and subtotals are cleared and don't influence backend calculations.

Reply:
Ah, makes sense! Yeah, I'm all for just sending up an ID and a quantity as our input. Factors in to the line/cart-level discount/taxes discussion as well.